sympy/core/function.py

```diff
@@ -1556,9 +1556,7 @@ def _hashable_content(self):
 
     def _eval_subs(self, old, new):
         if old in self.variables:
```

Comment:
Something that's not really handled correctly here is that `self.variables` can be any kind of expression, like `Subs(f(x), x*y, 0)`. I'm not completely sure what the correct logic should be, especially for multiple Subs (although I guess the best way to think about that is to think about them as nested Subs). 

But you don't need to fix this in this pull request if you don't want to. The fix here is already an improvement.